src/lib/Microsoft.Health.Events/EventConsumers/Service/EventConsumerService.cs

```diff
@@ -35,44 +51,42 @@ public async Task ConsumeEvents(IEnumerable<IEventMessage> events)
             {
                 foreach (IEventConsumer eventConsumer in _eventConsumers)
                 {
-                    await OperationWithRetryAsync(eventConsumer, events);
+                    try
+                    {
+                        await RetryPolicy.ExecuteAsync(async () => await eventConsumer.ConsumeAsync(events).ConfigureAwait(false));
+                    }
+                    catch (Exception e)
+                    {
+                        _logger.LogError(e);
+                    }
                 }
             }
         }
 
-        private async Task OperationWithRetryAsync(IEventConsumer eventConsumer, IEnumerable<IEventMessage> events)
+        private static AsyncPolicy CreateRetryPolicy(ITelemetryLogger logger)
         {
-            int currentRetry = 0;
-            double backoffMs = 0;
-            Random random = new Random();
-            bool operationComplete = false;
-
-            while (!operationComplete)
+            bool ExceptionRetryableFilter(Exception ee)
             {
-                try
+                switch (ee)
                 {
-                    if (currentRetry > 0 && backoffMs < _maximumBackoffMs)
-                    {
-                        int randomMs = random.Next(0, 1000);
-                        backoffMs = Math.Pow(2000, currentRetry) + randomMs;
-                        await Task.Delay((int)backoffMs);
-                    }
-
-                    await TryOperationAsync(eventConsumer, events).ConfigureAwait(false);
-                    break;
-                }
-#pragma warning disable CA1031
-                catch (Exception e)
-#pragma warning restore CA1031
-                {
-                    _logger.LogError(e);
+                    case AggregateException ae when ae.InnerExceptions.All(ExceptionRetryableFilter):
+                    case OperationCanceledException _:
+                    case HttpRequestException _:
+                    case EventHubsException _:
+                    case AuthenticationFailedException _:
+                    case RequestFailedException _:
+                        break;
+                    default:
+                        return false;
                 }
+
+                logger.LogError(new Exception("Encountered retryable exception", ee));
```

Comment:
We should emit telemetry here.

Reply:
Right now this would log the exception using the AI SDK, which would get recored as both an Exception Metric and Exception record. Are you referring to logging a different metric? If so, what metric would you like to see?

Reply:
For FHIR conversion we have explicit error codes for various non-recoverable errors.  See https://github.com/microsoft/iomt-fhir/blob/master/src/lib/Microsoft.Health.Fhir.Ingest/Telemetry/ExceptionTelemetryProcessor.cs and here is an example of a metric that emitted by one these exceptions https://github.com/microsoft/iomt-fhir/blob/master/src/lib/Microsoft.Health.Fhir.Ingest/Service/PatientDeviceMismatchException.cs

For this work item it is too large a scope to classify the various template errors we can see the same way but it would be good to make sure all the errors are getting tagged as general normalization errors.  If that is occurring upstream, great but I suspect it isn't.  Will may be able to help with more details since he did a lot of the wiring here to support Azure Monitor.

Reply:
I think that is fine, but agree that is out of scope for this task. When exceptions are thrown during Normalization they should wrapped in a `NormalizationException` and perhaps even implement `ITelemetryFormattable` to be turned into a metric. 
I can create a backlog item to capture that work.

Reply:
For clarification, the ask for this PR is ensure the metrics for errors thrown here have the following attributes:
``
        public Metric ToMetric => new Metric(
            "NormalizationException",
            new Dictionary<string, object>
            {
                { DimensionNames.Name, "NormalizationException" },
                { DimensionNames.Category, Category.Errors },
                { DimensionNames.ErrorType, ErrorType.DeviceMessageError },
                { DimensionNames.ErrorSeverity, ErrorSeverity.Warning },
                { DimensionNames.Operation, ConnectorOperation.Normalization},
            });
``

This should at least distinguish these errors in custom facing telemetry such that they can identify there is an issue with the device messages.  The exact nature will not be known, but that is what the follow up work item will seek to address.

Reply:
Spoke with @dustinburson  offline. We'll add in a custom ExceptionTelemetry processor to this class. The Normalization Service can use that to produce custom telemetry for each processed exception.